pkg/validation/projects.go

```diff
@@ -0,0 +1,59 @@
+package validation
+
+import (
+	"fmt"
+
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+)
+
+// ValidateStarterProjects checks if starter project has only one remote configured
+// and if the checkout remote matches the renote configured
+func ValidateStarterProjects(starterProject v1alpha2.StarterProject) (err error) {
+
+	var gitSource v1alpha2.GitLikeProjectSource
+	if starterProject.Git != nil {
+		gitSource = starterProject.Git.GitLikeProjectSource
+	} else if starterProject.Github != nil {
+		gitSource = starterProject.Github.GitLikeProjectSource
+	} else {
+		return
+	}
+
+	if len(gitSource.Remotes) != 1 {
+		return fmt.Errorf("starterProject can have only one remote")
```

Comment:
Consider splitting this into a zero remotes and a 2+ remotes case.

Reply:
The schema says this is mandatory, so that will have validated the zero case already 🤔

Reply:
In that case I think it'd be clearer if the check was `> 1`, but this is so minor that it's up to you if you want to change it.